app/src/main/java/com/bignerdranch/android/criminalintent/CrimeFragment.kt

```diff
@@ -41,6 +51,19 @@ class CrimeFragment: Fragment() {
         return view
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        crimeDetailViewModel.crimeLiveData.observe(
+            viewLifecycleOwner,
+            Observer { crime ->
+                crime?.let {
+                    this.crime = crime
```

Comment:
```
21. private lateinit var crime: Crime
...
33. crimeDetailViewModel.loadCrime(crimeId)
...
60. this.crime = crime
```

I assume you are doing this because you are following the book. However, this may be a great choice for a challenge.

Think about it - `ViewModel`s exist so they can keep the data that is then rendered by the UI (`Fragment` in your case). You already have a `LiveData` in your `ViewModel` that returns a `Crime` object. Therefore, you _don't need to_ hold any reference to a `Crime` object in here, in the `Fragment`, because you can simply ask your viewmodel to provide it, whenever you need to. So, because of that, you if you think about it, you will see that you don't need to pass an argument to your `viewModel.saveCrime()` function, as you do in here on line `107` 🙂.

This not only is a good practice, but also will save you from weird bugs known as [race conditions](https://stackoverflow.com/questions/34510/what-is-a-race-condition).